fenicsxconcrete/finite_element_problem/base_material.py

```diff
@@ -109,6 +109,11 @@ def solve(self):
         # define what to do, to solve this problem
         pass
 
+    @abstractmethod
+    def compute_residuals(self):
+        # define what to do, to compute the residuals. Called in solve
+        pass
```

Comment:
abstract method does not require pass (it reduces our coverage), best just to add a `""" comment """`

Reply:
Changed, as well as others that were there. Maybe it conflicts with another branch if it is already updated, but should be straightforward to merge.